worlds/stardew_valley/regions.py

```diff
@@ -353,8 +354,81 @@ def inverted(self):
 ]
 
 
+# Attempt to implement remote loading of modded data if relevant (to avoid circular references and clutter)
+def loaded_mod_region_data(mod: str, data_type: str, world_options: StardewOptions):
+    loaded_mod_data = []
+    if mod in world_options[options.Mods]:
+        for mod_data in ModDataList:
+            if (mod_data.mod_name == mod) & (data_type == "Regions"):
+                for region in mod_data.regions:
+                    if not mod_data.regions[region]:
+                        loaded_mod_data.append(RegionData(region))
+                        continue
+                    loaded_mod_data.append(RegionData(region, mod_data.regions[region]))
+            if (mod_data.mod_name == mod) & (data_type == "Connections"):
+                for entrance in mod_data.connections:
+                    if mod_data.flags[entrance] == "NON_RANDOMIZED":
```

Comment:
The mod flags should be using the actual flags, not strings that are based on the flag names.

Here, this would allow gettind rid of all of these ifs, due to you simply being able to match the flag directly